src/dotnet/CommandDirectoryContext.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.IO;
+
+namespace Microsoft.DotNet.Cli
+{
+    internal static class CommandDirectoryContext
+    {
+        [ThreadStatic]
+        private static string _currentDirectoryOverwrite;
+
+        /// <summary>
+        /// Expands a path similar to Path.GetFullPath() but allows unit tests a hook to inject an overwrite to the
+        /// current working directory.
+        /// </summary>
+        /// <param name="path">A realtive or absolute path specifier</param>
+        /// <returns>The full path to the target</returns>
+        public static string ExpandPath(string path)
```

Comment:
Nit: I think this should be named `GetFullPath` as well to better convey the fact that this is just `Path.GetFullPath` with a configurable base path.